native/jni/magiskboot/bootimg.cpp

```diff
@@ -761,7 +761,11 @@ void repack(const char *src_img, const char *out_img, bool skip_comp) {
         memcpy(footer, boot.avb_footer, sizeof(AvbFooter));
         footer->original_image_size = __builtin_bswap64(off.total);
         footer->vbmeta_offset = __builtin_bswap64(off.vbmeta);
-        vbmeta->flags = __builtin_bswap32(3);
+        if(keep_dm_verity) {
+            vbmeta->flags = __builtin_bswap32(0);
+        } else {
+            vbmeta->flags = __builtin_bswap32(3);
+        }
```

Comment:
I would replace:
```c++
        if(keep_dm_verity) {
            vbmeta->flags = __builtin_bswap32(0);
        } else {
            vbmeta->flags = __builtin_bswap32(3);
        }
```
By
```c++
        if(!keep_dm_verity) {
            vbmeta->flags = __builtin_bswap32(3);
        }
```
to avoid conflicting with custom ROMs that may require 3 as vbmeta flag.
(All stock ROMs are supposed to have a 0 `vbmeta->flags` by default)